src/Stateless/Graph/UmlDotGraphStyle.cs

```
@@ -0,0 +1,179 @@
+                    }
+                    else if (stay.SourceState.EntryActions.Count() == 0)
+                    {
+                        line = FormatOneTransition(stay.SourceState.NodeName, stay.Trigger.UnderlyingTrigger.ToString(),
+                            null, stay.SourceState.NodeName, stay.Guards.Select(x => x.Description));
+                    }
+                    else
+                    {
+                        // There are entry functions into the state, so call out that this transition
+                        // does invoke them (since normally a transition back into the same state doesn't)
+                        line = FormatOneTransition(stay.SourceState.NodeName, stay.Trigger.UnderlyingTrigger.ToString(),
+                            stay.SourceState.EntryActions, stay.SourceState.NodeName, stay.Guards.Select(x => x.Description));
+                    }
+                }
+                else
+                {
+                    if (transit is FixedTransition fix)
+                    {
+                        line = FormatOneTransition(fix.SourceState.NodeName, fix.Trigger.UnderlyingTrigger.ToString(),
+                            fix.DestinationEntryActions.Select(x => x.Method.Description),
+                            fix.DestinationState.NodeName, fix.Guards.Select(x => x.Description));
+                    }
+                    else
+                    {
+                        if (transit is DynamicTransition dyn)
+                        {
+                            line = FormatOneTransition(dyn.SourceState.NodeName, dyn.Trigger.UnderlyingTrigger.ToString(),
+                                dyn.DestinationEntryActions.Select(x => x.Method.Description),
+                                dyn.DestinationState.NodeName, new List<string> { dyn.Criterion });
+                        }
+                        else
+                            throw new System.Exception("Unexpected transition type");
```

Could we use a more specific exception type? If you throw such a general exception type in a library or framework, it forces consumers to catch all exceptions, including unknown exceptions that they do not know how to handle.